src/Chunk.cpp

```diff
@@ -1570,6 +1570,10 @@ void cChunk::FastSetBlock(int a_RelX, int a_RelY, int a_RelZ, BLOCKTYPE a_BlockT
 	if (
 		a_SendToClients &&                  // ... we are told to do so AND ...
 		(
+			!(                                // ... the old and new blocktypes AREN'T leaves (because the client doesn't need meta updates)
+				((OldBlockType == E_BLOCK_LEAVES) && (a_BlockType == E_BLOCK_LEAVES)) ||
+				((OldBlockType == E_BLOCK_NEW_LEAVES) && (a_BlockType == E_BLOCK_NEW_LEAVES))
+			) ||                              // ... OR ...
```

Comment:
Doesn't this need to be an and sat the end? This would cause the server to alway update blocks that aren't leaves rather that not updating leaves on meta changes.